exec-service-wasmer/src/wasmer_metering.rs

```diff
@@ -62,11 +68,18 @@ impl ModuleMiddleware for Metering {
         Box::new(FunctionMetering {
             accumulated_cost: Default::default(),
             opcode_cost: self.opcode_cost.clone(),
-            global_indexes: self.global_indexes.borrow().clone().unwrap(),
+            breakpoint: self.breakpoint.clone(),
+            global_indexes: self.global_indexes.lock().unwrap().clone().unwrap(),
         })
     }
 
     fn transform_module_info(&self, module_info: &mut ModuleInfo) {
+        let mut global_indexes = self.global_indexes.lock().unwrap();
+
+        if global_indexes.is_some() {
+            panic!("Metering::transform_module_info: Attempting to use a `Metering` middleware from multiple modules.");
```

Comment:
Why is this necessary?

Reply:
Not necessary in our case. We can remove it.